api/src/main/kotlin/com/google/prefab/api/Android.kt

```diff
@@ -263,6 +263,77 @@ class Android(val abi: Abi, api: Int, val stl: Stl, val ndkMajorVersion: Int) :
         return true
     }
 
+    override fun findBestMatch(
+        libraries: List<PrebuiltLibrary>
+    ): PrebuiltLibrary {
+        require(libraries.isNotEmpty())
+        require(libraries.all { canUse(it) })
+        val moduleName = libraries.first().module.canonicalName
+
+        val allLibraries: List<Pair<PrebuiltLibrary, Android>> = libraries.map {
+            require(it.platform is Android)
+            Pair(it, it.platform)
+        }
+
+        // Filter out any libraries that were built with a lower API level.
+        // Libraries built for newer API levels may expose additional features
+        // the rely on system APIs not available in earlier API levels, and may
+        // be smaller since libandroid_support is only required on very old API
+        // levels.
+        val bestApiLevel = allLibraries.maxBy { it.second.api }!!.second.api
+        val bestApiLevelMatches = allLibraries.filter { (_, reqs) ->
+            reqs.api == bestApiLevel
+        }
+
+        if (bestApiLevelMatches.size == 1) {
+            return bestApiLevelMatches.single().first
```

Comment:
is it deliberate that if there's only a single match, you'll return it even if it's for an unsuitable NDK?

Reply:
Yes. I suspect very few packages will actually provide a library per-NDK version. I'll add a comment saying that.